moveit_ros_control_interface/src/controller_manager_plugin.cpp

```diff
@@ -0,0 +1,423 @@
+        boost::mutex::scoped_lock lock(controllers_mutex_);
+        discover(true);
+
+        typedef boost::bimap< std::string, std::string> resources_bimap;
+
+        resources_bimap claimed_resources;
+
+        // fill bimap with active controllers and their resources
+        for(ControllersMap::iterator c = active_controllers_.begin(); c != active_controllers_.end(); ++c){
+            for(std::vector<std::string>::iterator r = c->second.resources.begin(); r != c->second.resources.end(); ++r){
+                claimed_resources.insert(resources_bimap::value_type(c->second.name, *r));
+            }
+        }
+
+        controller_manager_msgs::SwitchController srv;
+
+        for(std::vector<std::string>::const_iterator it = deactivate.begin(); it != deactivate.end(); ++it){
+            ControllersMap::iterator c = managed_controllers_.find(*it);
+            if(c != managed_controllers_.end()){ // controller belongs to this manager
+                srv.request.stop_controllers.push_back(c->second.name);
+                claimed_resources.right.erase(c->second.name); // remove resources
+            }
+        }
+
+
+        for(std::vector<std::string>::const_iterator it = activate.begin(); it != activate.end(); ++it){
+            ControllersMap::iterator c = managed_controllers_.find(*it);
+            if(c != managed_controllers_.end()){ // controller belongs to this manager
+                srv.request.start_controllers.push_back(c->second.name);
+
+                for(std::vector<std::string>::iterator r = c->second.resources.begin(); r != c->second.resources.end(); ++r){ // for all claimed resource
+                    resources_bimap::right_const_iterator res = claimed_resources.right.find(*r);
+                    if(res != claimed_resources.right.end()){ // resource is claimed
+                        srv.request.stop_controllers.push_back(res->second); // add claiming controller to stop list
+                        claimed_resources.left.erase(res->second);  // remove claimed resources
+                    }
+                }
+            }
+        }
+        srv.request.strictness = srv.request.STRICT;
+
+        if (!srv.request.start_controllers.empty() || srv.request.stop_controllers.empty()){ // something to switch?
+            if(!ros::service::call(ns_ + "controller_manager/switch_controller", srv)){
+                ROS_ERROR_STREAM("Could switch controllers at " << ns_);
+            }
+            discover(true);
+            return srv.response.ok;
+        }
+        return true;
+    }
+    typedef boost::shared_ptr<MoveItControllerManager> Ptr;
+};
+/**
+ * MoveItMultiControllerManager discovers all running ros_control node and delegates member function to the corresponding MoveItControllerManager instances
+ */
+class  MoveItMultiControllerManager : public moveit_controller_manager::MoveItControllerManager {
```

Comment:
How would the usability be improved if they are in different files?
Both share the `checkTimeout` helper, but this is no blocker.